WordPress/src/main/java/org/wordpress/android/widgets/WPViewPager.java

```diff
@@ -5,45 +5,51 @@
 import android.util.AttributeSet;
 import android.view.MotionEvent;
 
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+
+/**
+ * Custom ViewPager which resolves the "pointer index out of range" bug in the compatibility library
+ * https://code.google.com/p/android/issues/detail?id=16836
+ * https://code.google.com/p/android/issues/detail?id=18990
+ * https://github.com/chrisbanes/PhotoView/issues/31
+ */
 public class WPViewPager extends ViewPager {
-    private boolean enabled;
-    private int mPreviousPage;
+    private boolean mPagingEnabled = true;
+
+    public WPViewPager(Context context) {
+        super(context);
+    }
 
     public WPViewPager(Context context, AttributeSet attrs) {
         super(context, attrs);
-        this.enabled = false;
-        mPreviousPage = 0;
     }
 
     @Override
-    public boolean onTouchEvent(MotionEvent event) {
-        if (this.enabled) {
-            return super.onTouchEvent(event);
+    public boolean onInterceptTouchEvent(MotionEvent ev) {
+        if (mPagingEnabled) {
+            try {
+                return super.onInterceptTouchEvent(ev);
+            } catch (IllegalArgumentException e) {
+                AppLog.e(T.UTILS, e);
+            }
         }
-
         return false;
     }
 
     @Override
-    public boolean onInterceptTouchEvent(MotionEvent event) {
-        if (this.enabled) {
-            return super.onInterceptTouchEvent(event);
+    public boolean onTouchEvent(MotionEvent ev) {
+        if (mPagingEnabled) {
+            try {
+                return super.onTouchEvent(ev);
+            } catch (IllegalArgumentException e) {
+                AppLog.e(AppLog.T.UTILS, e);
+            }
         }
-
         return false;
     }
 
-    public void setPagingEnabled(boolean enabled) {
-        this.enabled = enabled;
-    }
-
-    @Override
-    public void setCurrentItem(int currentItem) {
-        mPreviousPage = getCurrentItem();
-        super.setCurrentItem(currentItem);
-    }
-
-    public int getPreviousPage() {
-        return mPreviousPage;
+    public void setPagingEnabled(boolean pagingEnabled) {
+        mPagingEnabled = pagingEnabled;
```

Comment:
`mPagingEnabled` is used [here](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2848-viewpager-merge/WordPress/src/main/java/org/wordpress/android/widgets/WPViewPager.java#L30) and [here](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2848-viewpager-merge/WordPress/src/main/java/org/wordpress/android/widgets/WPViewPager.java#L42) - and it's used in the MediaPicker and the Editor.